### PR TITLE
[Sig-Security] Generation of SLSA3+ provenance for KubeEdge release artifacts

### PR DESCRIPTION
**Pull request by vincentgoat**
Signed-off-by: vincentgoat <linguohui1@huawei.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Generation of SLSA3+ provenance for KubeEdge, generating non-forgeable [SLSA provenance](https://slsa.dev/) on GitHub that meets the [build](https://slsa.dev/spec/v0.1/requirements#build-requirements) and [provenance](https://slsa.dev/spec/v0.1/requirements#provenance-requirements) requirements for [SLSA level 3 and above](https://slsa.dev/spec/v0.1/levels).

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**: **YES**

### 1. Before merging this PR
Please create a `Repository variables` in the repo kubeedge.
**Steps:** Repo kubeedge > settings > Security/Secretes and variables > Actions > Variables > New repositories variables
New the variable: `DOCKERHUB_USER_NAME: {the value is same with dockerhub username}`

### 2. Here is how we verify provenance generated by SLSA generator (Binary artifacts):

1. Download slsa-verifier tools. FYI: https://github.com/slsa-framework/slsa-verifier

2. Example:

    $ **./slsa-verifier-linux-amd64 verify-artifact kubeedge-v1.xx.xx-linux-arm.tar.gz --provenance-path attestation.intoto.jsonl --source-uri github.com/kubeedge/kubeedge --source-tag v1.xx.xx**

If the results return the following means verifying successfully.
`PASSED: Verified SLSA provenance`

Otherwise failed.
`expected hash 'xxx' not found: artifact hash does not match provenance subject`

**Note:** Files `kubeedge-v1.xx.xx-linux-arm.tar.gz` and `attestation.intoto.jsonl` are release artifacts that show at KubeEdge RELEASE pages.

### 3. Verify container images' provenance
FYI: https://github.com/slsa-framework/slsa-github-generator/tree/main/internal/builders/container#verification

Examples:
$ COSIGN_EXPERIMENTAL=1 cosign verify-attestation --type slsaprovenance --policy policy.cue kubeedge/admission:v1.x.x

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -20,8 +20,6 @@ jobs:
           - linux
         arch:
```
Comment (gy95) [inline .github/workflows/release.yml:21]: delete this arch part

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -37,15 +35,70 @@ jobs:
...
         if: ${{ !env.ACT }}
         uses: softprops/action-gh-release@v1
         with:
           files: |
-            _output/release/${{ github.ref_name }}/${{ matrix.target }}-${{ github.ref_name }}-${{ matrix.os }}-${{ matrix.arch }}.tar.gz
-            _output/release/${{ github.ref_name }}/checksum_${{ matrix.target }}-${{ github.ref_name }}-${{ matrix.os }}-${{ matrix.arch }}.tar.gz.txt
-
+            _output/release/${{ github.ref_name }}/${{ matrix.target }}-${{ github.ref_name }}-${{ matrix.os }}-${{ env.output_arch }}.tar.gz
+            _output/release/${{ github.ref_name }}/checksum_${{ matrix.target }}-${{ github.ref_name }}-${{ matrix.os }}-${{ env.output_arch }}.tar.gz.txt
+  build-hash:
```
Comment (gy95) [inline .github/workflows/release.yml:69]: add a blank line between the two  jobs and between the two actions to make it readable

File: .github/workflows/release.yml (hunk 1 above)
Comment (vincentgoat) [reply .github/workflows/release.yml]: done

File: .github/workflows/release.yml (hunk 2 above)
Comment (vincentgoat) [reply .github/workflows/release.yml:68]: done

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**